### ISSUE TITLE
what are `inference and training outputs` when model is in `eval` mode?

### ISSUE THREAD

**Issue by FraPochetti** [ultralytics/yolov5#2546]
## ❔Question
Hi, 
I noticed that when the model is in `train` mode, it returns just `pred` (as expected - [here](https://github.com/ultralytics/yolov5/blob/ed2c74218d6d46605cc5fa68ce9bd6ece213abe4/train.py#L296)).
Whereas, when it is in `eval` mode, it returns 2 outputs `out, train_out` ([here](https://github.com/ultralytics/yolov5/blob/ed2c74218d6d46605cc5fa68ce9bd6ece213abe4/test.py#L110)), commented with `# inference and training outputs`. Inspecting those, it seems `train_out` contains the actual predictions. What is `out` then?

Thanks and have a great one!

**Comment by glenn-jocher** [comment]
@FraPochetti reverse. out is inference output, train_out is training outputs. During training only training outputs are used.

**Comment by FraPochetti** [comment]
Got it thanks a lot!

**Comment by glenn-jocher** [comment]
@FraPochetti no problem! The way to think of this is that the training outputs are pre-sigmoid and are used to generate loss, while the inference outputs are fully processed and intended for NMS.

**Comment by FraPochetti** [comment]
Indeed! Thanks a lot, Glenn. Super useful as always!

**Comment by FraPochetti** [comment]
@glenn-jocher now that I have your attention :D I have another (hopefully quick) question.
I looked carefully at the data preparation (both dataset and dataloader) steps and I couldn't find a place where you normalize the images using ImageNet stats. I only spotted the `div(255)` to squash inputs in the `[0, 1]` range.
Is this intended?
Of course this makes just for the pretrained models (if they were pretrained on ImageNet at all).

**Comment by glenn-jocher** [comment]
@FraPochetti that's right, there's no mean and std normalization, just a rescaling from 0-255 to 0-1 range. We tried mean subtraction and std division, but this did not improve model results. 

I think the reason an additional normalizations step has no effect (on COCO at least) is that the initial batchnorm layers are effectively doing this same task.

**Comment by FraPochetti** [comment]
good point @glenn-jocher.
 
btw man, you are a true gift to humanity!
thanks for sharing your knowledge and for your patience!
